ui/src/app/utils/index.ts

```diff
@@ -6,3 +6,7 @@ export function isBlank(toTest: String): boolean {
     return toTest === '';
   }
 }
+
+export function deepCopy(obj: Object): Object {
```

Comment:
Just be aware that this will clobber function-valued keys and dates - functions have no JSON repr and dates will be converted to strings and then stay that will.  In general this is only a good idea if all the values of the object are JSON primitives.

It _is_ a fast and easy way to clone (simple) objects but we probably want a comment with the relevant caveats to prevent people from thinking this is a good idea for objects that may have complex values; maybe even name it `unsafeDeepCopy`.  Alternatively (but slower) we can use `immutable.js`, which is already available.  It would look something like:

```
return fromJS(object).toJS();
```

That should give you a copy without altering the original object and without losing any information but it'll be slower (probably, I didn't benchmark it or anything).

Reply:
I'd actually much rather have us have a slower deepcopy. Will move to this.